featuretools/primitives/standard/transform/numeric/rate_of_change.py

```diff
@@ -0,0 +1,34 @@
+from woodwork.column_schema import ColumnSchema
+from woodwork.logical_types import Double
+
+from featuretools.primitives.base import TransformPrimitive
+from featuretools.utils.gen_utils import Library
+
+
+class RateOfChange(TransformPrimitive):
+    """Computes the rate of change of a value per second.
+
+    Examples:
+        >>> import pandas as pd
+        >>> rate_of_change = RateOfChange()
+        >>> times = pd.date_range(start='2019-01-01', freq='1min', periods=5)
+        >>> rate_of_change([0, 30, 180, -90, 0], times).tolist()
+        [nan, 0.5, 2.5, -4.5, 1.5]
+    """
+
+    name = "rate_of_change"
+    input_types = [
+        ColumnSchema(semantic_tags={"numeric"}),
+        ColumnSchema(semantic_tags={"time_index"}),
+    ]
+    return_type = ColumnSchema(logical_type=Double, semantic_tags={"numeric"})
+    uses_full_dataframe = True
+    description_template = "the rate of change of of {}"
+
+    def get_function(self):
+        def rate_of_change(values, time):
+            time_delta = time.diff().dt.total_seconds()
+            value_delta = values.diff()
+            return value_delta / time_delta
```

Comment:
Curious, what is the difference between your approach and this pandas function:
- https://pandas.pydata.org/docs/reference/api/pandas.DataFrame.pct_change.html

Reply:
Based on a quick read of the documentation, that function appears to just be calculating a percent change between consecutive values. This primitive calculates the rate of change between points (think velocity).

Take this dataframe as an example: 
```
values = pd.Series([0, 30, 180, -90, 0, 240])
time = pd.Series(pd.date_range(start='2019-01-01', freq='1min', periods=6))
df = pd.DataFrame({"values": values, "time": time})
```

If you use the pandas `pct_change` method you get this:
```
0    NaN
1    inf
2    5.0
3   -1.5
4   -1.0
5    inf
```

If you compute the rate of change using the elapsed time between points from the time index you get this:
```
0    NaN
1    0.5
2    2.5
3   -4.5
4    1.5
5    4.0
```

Reply:
If the values were a distance measurement recorded in `meters`, the pandas function just tells you how much a value increased or decreased as a percentage of the previous one (`%`). This primitive will give you the rate of change of distance, which is a velocity in this case measured in units of `meters per second`.

Said another way, this primitive is basically giving you the first derivative of a set of values that are ordered by time.